### DISCUSSION TITLE
2021-12-17 Core Tools Update

### DISCUSSION THREAD

**Discussion by austinabell** [near/community#154] [Developer Tools]
### near-sdk-rs
- @itegulov WIP new `UnorderedSet` implementation (https://github.com/near/near-sdk-rs/pull/672)
- @austinabell Update `LookupMap` key traits to allow being backwards compatible with previous format (https://github.com/near/near-sdk-rs/pull/674)
- @austinabell Update [to 2021 Rust edition](https://github.com/near/near-sdk-rs/pull/669) and [prepare repo for stabilization](https://github.com/near/near-sdk-rs/pull/671) 
- @itegulov Added new collections benchmarks to compare `near_sdk::collections` with `near_sdk::store` for new structures (https://github.com/itegulov/storage_collections_bench/tree/set-benchmarks)

### workspaces-rs
- 🎉🎉 @ChaoticTempest [Released](https://crates.io/crates/workspaces/versions) version `0.1` of workspaces. 🎉🎉
  - This includes various changes made to get to release, including input from community, team members, and other stakeholders
- @ChaoticTempest Merged ref-finance example for spooning and testing based on live state (https://github.com/near/workspaces-rs/pull/35)

### nearcore
- @miraclx Implemented various PRs for public crate compliance (https://github.com/near/nearcore/issues/5763)

**Comment by maxzaver** [comment]
> 🎉🎉 @ChaoticTempest Released version 0.1 of workspaces. 🎉🎉

@austinabell , @ChaoticTempest  Does crates.io `v0.1` correspond to Github `v0.1` https://github.com/near/workspaces-rs/releases/tag/v0.1 ?

Also, regarding near-sdk-rs v4.0.0 AFAIR it requires a protocol upgrade. Do we know which nearcore version will it be?

**Reply by austinabell** [reply]
> @austinabell , @ChaoticTempest Does crates.io `v0.1` correspond to Github `v0.1` [near/workspaces-rs@`v0.1` (release)](https://github.com/near/workspaces-rs/releases/tag/v0.1) ?
> 
> Also, regarding near-sdk-rs v4.0.0 AFAIR it requires a protocol upgrade. Do we know which nearcore version will it be?

No, that commit is not what was released. I'll update this.

4.0.0 only requires a protocol upgrade if it includes the NEP for utilizing unused gas w/ API changes. I've suggested that we stabilize 4.0 and have that change come with the next major version which can be iterated on in parallel.
